app/controllers/queue_items_controller.rb

```diff
@@ -0,0 +1,40 @@
+class QueueItemsController < ApplicationController
+
+  before_action :authorize
+
+  def index
+    @queue_items = current_user.queue_items
+  end
+
+  def create
+    queue_item = QueueItem.new(queue_items_params)
+    if queue_item.save
+      flash[:notice] = "Video added to your queue!"
+    else
+      flash[:alert] = "Can't add that video to your queue!"
+    end
+    redirect_to queue_items_path
+  end
+
+  def destroy
+    queue_item = QueueItem.find(params[:id])
+    if current_user.queue_items.include?(queue_item)
+      queue_item.destroy
+      redirect_to queue_items_path
+    else
+      flash[:alert] = "Please sign in!"
```

Comment:
Is this the only time someone can not have the video queued? You are also redirecting to sign in, but I assume that the `before_action :authorize` already handles that case.